catalogue/src/main/kotlin/com/tidal/sdk/catalogue/generated/infrastructure/AtomicBooleanAdapter.kt

```diff
@@ -0,0 +1,21 @@
+package com.tidal.sdk.catalogue.generated.infrastructure
+
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.Serializer
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
+import kotlinx.serialization.descriptors.PrimitiveKind
+import kotlinx.serialization.descriptors.SerialDescriptor
+import java.util.concurrent.atomic.AtomicBoolean
+
+@Serializer(forClass = AtomicBoolean::class)
+object AtomicBooleanAdapter : KSerializer<AtomicBoolean> {
```

Comment:
you really should never serialize `AtomicBoolean` since our data classes should be immutable and `AtomicBoolean` is an object that is made for thread-safe mutation. Any reason we are doing that?

Edit: I just realized this is in the generated package. Why are they generating such senseless serializers? Shall we delete them?

Reply:
I have to go through them anyways since the serializers are also blocking us from backwards compatibility.
I have not yet tested if they actually are all used with our models or if they are generated regardless.